src/natstime.c

```diff
@@ -58,6 +58,33 @@ nats_NowInNanoSeconds(void)
 #endif
 }
 
+int64_t
+nats_NowMonotonicInNanoSeconds(void)
+{
+#ifdef _WIN32
+    LARGE_INTEGER frequency;
+    LARGE_INTEGER counter;
+    if (!QueryPerformanceFrequency(&frequency) || !QueryPerformanceCounter(&counter))
+        abort();
+    return (int64_t)(counter.QuadPart * 1000000000ULL / frequency.QuadPart);
+#elif defined CLOCK_MONOTONIC
+    struct timespec ts;
+    if (clock_gettime(CLOCK_MONOTONIC, &ts) != 0)
+        abort();
+    return ((int64_t) ts.tv_sec) * 1000000000L + ((int64_t) ts.tv_nsec);
+#elif defined CLOCK_REALTIME
```

Comment:
Anything else, I would simply return `nats_NowInNanoSeconds()` instead.

So you could rewrite the function so that we have an int64 value at the beginning set to 0, and if win32 and query API works, set the int64 value, if not windows but clock_monotonic defined (and returns 0), then set the int64. Then at the end of the function, if value is still 0, then set to `nats_NowInNanoSeconds()` and the end of function simply returns the int64 value. Does that make sense?